Steps4Impact/NotificationsV2/NotificationsViewController.swift

```diff
@@ -31,35 +31,96 @@ import UIKit
 import RxSwift
 
 class NotificationsViewController: TableViewController {
+  
   override func commonInit() {
     super.commonInit()
 
     title = Strings.Notifications.title
     dataSource = NotificationsDataSource()
   }
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    checkNotificationPermission()
+  }
+  
+  override func reload() {
+    dataSource?.reload { [weak self] in
+      self?.tableView.reloadOnMain()
+    }
+  }
+  
+  override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+    super.tableView(tableView, willDisplay: cell, forRowAt: indexPath)
+    if let cell = cell as? NotificationPermissionCell {
+      cell.delegate = self
+    }
+  }
+  
+  func checkNotificationPermission() {
+    func reloadIfNeeded() {
+      guard let dataSource = dataSource as? NotificationsDataSource else {
+        return
+      }
+      if let _ = dataSource.cells.first as? [NotificationPermissionCellContext] {
+         if UIApplication.shared.isRegisteredForRemoteNotifications {
+            dataSource.cells.remove(at: 0)
+            tableView.reloadOnMain()
+         }
+      } else if !UIApplication.shared.isRegisteredForRemoteNotifications, dataSource.canShowPrompt() {
+        dataSource.cells.insert([dataSource.notificationPermissionCellContext], at: 0)
+        tableView.reloadOnMain()
+      }
+    }
+    
+    reloadIfNeeded()
+  }
 }
 
 class NotificationsDataSource: TableViewDataSource {
   let cache = Cache.shared
   var cells: [[CellContext]] = []
   var disposeBag = DisposeBag()
   
+  let notificationPermissionCellContext = NotificationPermissionCellContext(title: Strings.NotificationsPermission.title,
+                                                                            description: Strings.NotificationsPermission.message,
+                                                                            disclosureText: Strings.NotificationsPermission.discloreText)
+  
   init() {
     
     cache.participantRelay.subscribeOnNext { [weak self] (participant: Participant?) in
       guard let fbId = participant?.fbid, let eventId = participant?.currentEvent?.id else { return }
       
       AKFCausesService.getNotifications(fbId: fbId, eventId: eventId) { (result) in
         print(result)
```

Comment:
Will hook-up the backend response in a different task. Have a ticket for that - https://github.com/AgaKhanFoundation/WCF-iOS/issues/375